---
title: "A02. Engaging new and experienced users"
author:
- name: Martin Morgan
  affiliation: Roswell Park Comperhensive Cancer Center, Buffalo, NY
date: September 27 2018
output:
  BiocStyle::html_document:
    toc_float: true
vignette: |
  %\VignetteIndexEntry{A02. Engaging new and experienced users}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'asis'}
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE"))
)
```

# File-based data formats pervade bioinformatics

Bioinformatics data often comes in particular file formats

- FASTA: DNA or RNA sequence(s)
- FASTQ: short DNA reads and their quality scores, from
  high-throughput sequencing
- SAM / BAM: aligned reads
- GFF / GTF: gene models
- BED: genomic annotation 'tracks'
- ...

Files are often 'easy' to read using base _R_ commands, but they often
have subtle issues. For instance, BED files describe genomic ranges
using 0-based, half-open intervals whereas other file formats use
1-based, closed intervals or do not specify how genomic coordinates
are represented.

# _Bioconductor_ represents data in standardized ways

_Bioconductor_ software tries to adopt a consistent scheme for
representing genomic ranges -- 1-based, closed intervals. 

The `GenomicRanges` class emphasizes this standardized structure, and
enables interoperability between packages.

Here is a BED file containing peaks from a ChIP-seq experiment.

```{r}
path <- "~/extdata/chipseq/GSM1174480_ARmo_0M_peaks.bed.gz"
```

It is a plain-old tab-delimited file with three columns

```{r}
readLines(path, 3)
```

Import the BED file into _R_ using the [rtracklayer][] package
`import()` function.

```{r, message=FALSE}
library(rtracklayer)
bed <- import(path)
```

Display the object, looking closely at the information... How have
individual ranges been standardized?

```{r}
bed
```

[rtracklayer]: https://bioconductor.org/packages/rtracklayer

# Many packages provide compact work flows for standard analysis

[ChIPseeker][] vignette

- Urbanucci A, Sahu B, Seppälä J, Larjo A et al. Overexpression of
  androgen receptor enhances the binding of the receptor to the
  chromatin in prostate cancer. Oncogene 2012 Apr
  26;31(17):2153-63. PMID: 21909140. [GSE48308][]
  
[ChIPseeker]: https://bioconductor.org/packages/ChIPseeker
[GSE48308]: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48308

```{r, message=FALSE}
library(ChIPseeker)
```

```{r}
fls <- dir("~/extdata/chipseq", "*bed.gz", full = TRUE)
peak <- readPeakFile(fls[[4]])
peak
covplot(peak)
```

```{r, message=FALSE}
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
```

```{r}
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
tagMatrix <- getTagMatrix(peak, windows=promoter)
dim(tagMatrix)
plotAvgProf(tagMatrix, xlim=c(-3000, 3000))
```

```{r}
tagHeatmap(tagMatrix, xlim=c(-3000, 3000))
```

```{r}
anno <- annotatePeak(
    fls[[4]], tssRegion=c(-3000, 3000), TxDb=txdb, annoDb="org.Hs.eg.db"
)
anno
```

```{r}
plotAnnoPie(anno)
```

```{r}
upsetplot(anno)
```

```{r}
ChIPseeker::as.GRanges(anno)
```
