---
title: "A03. Practical activities with meaningful feedback"
author:
- name: Martin Morgan
  affiliation: Roswell Park Comperhensive Cancer Center, Buffalo, NY
date: September 27 2018
output:
  BiocStyle::html_document:
    toc_float: true
vignette: |
  %\VignetteIndexEntry{A03. Practical activities with meaningful feedback}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'asis'}
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE"))
)
```

[ChIPseeker][] vignette

- Urbanucci A, Sahu B, Seppälä J, Larjo A et al. Overexpression of
  androgen receptor enhances the binding of the receptor to the
  chromatin in prostate cancer. Oncogene 2012 Apr
  26;31(17):2153-63. PMID: 21909140. [GSE48308][]
  
  
[ChIPseeker]: https://bioconductor.org/packages/ChIPseeker
[GSE48308]: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48308

# _GRanges_ are an important component of the _Bioconductor_ ecosystem

```{r, message=FALSE}
library(rtracklayer)
dir("~/extdata/chipseq")
path <- "~/extdata/chipseq/GSM1174480_ARmo_0M_peaks.bed.gz"
```

```{r}
peak <- import(path)
peak
```

QUESTION

- Use `methods(class="GRanges")` to see (many of the) methods
  available for a `GRanges` object.

Manipulating each range, e.g.,

- `flank()` selects ranges up- or down- stream of individual ranges.

Comparing ranges in a single object, e.g.,

- `any(duplicated(bed))`

Comparing ranges between objects, e.g.,

- `nearest(x, subject)` for each range in `x`, find the index of the
  nearest range in `subject`.

# Objects and their methods provide building blocks for meaningful manipulation

```{r, message=FALSE}
library(GenomicFeatures)                   # software
library(TxDb.Hsapiens.UCSC.hg19.knownGene) # Gene and transcript models
```

```{r}
exons <- exons(TxDb.Hsapiens.UCSC.hg19.knownGene)
peak$exonic <- peak %over% exons
```

```{r}
peak$genic <- peak %over% genes(TxDb.Hsapiens.UCSC.hg19.knownGene)
```

QUESTION: what about peaks overlapping introns?

QUESTION: nearest transcripts?

```{r}
tx <- transcripts(TxDb.Hsapiens.UCSC.hg19.knownGene)
```

```{r}
nearest_tx <- tx[ nearest(peak, tx) ]
```

```{r, message=FALSE}
library(org.Hs.eg.db)                      # gene identifier maps
```

```{r}
peak$ENTREZID <- mapIds(
    TxDb.Hsapiens.UCSC.hg19.knownGene, nearest_tx$tx_name, "GENEID", "TXNAME"
)
idx <- !is.na(peak$ENTREZID)
peak$SYMBOL[idx] <- mapIds(
    org.Hs.eg.db, peak$ENTREZID[idx], "SYMBOL", "ENTREZID"
)
peak
```

# Your own functions capture complex transformations for easy reuse

```{r}
annotate_peaks <- function(path, txdb, orgdb) {
    peak <- import(path)

    peak$genic <- peak %over% genes(txdb)
    peak$exonic <- peak %over% exons(txdb)
    peak$intronic <- peak %over% intronsByTranscript(txdb)

    tx <- transcripts(txdb)
    nearest_tx <- tx[ nearest(peak, tx) ]
    peak$ENTREZID <- mapIds(txdb, nearest_tx$tx_name, "GENEID", "TXNAME")
    idx <- !is.na(peak$ENTREZID)
    peak$SYMBOL[idx] <- 
        mapIds(orgdb, peak$ENTREZID[idx], "SYMBOL", "ENTREZID")
    peak
}
```

```{r}
annotate_peaks(path, TxDb.Hsapiens.UCSC.hg19.knownGene, org.Hs.eg.db)
```
