---
title: "A05. Managing cognitive load"
author:
- name: Martin Morgan
  affiliation: Roswell Park Comperhensive Cancer Center, Buffalo, NY
date: September 27 2018
output:
  BiocStyle::html_document:
    toc_float: true
vignette: |
  %\VignetteIndexEntry{A05. Managing cognitive load}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'asis'}
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE"))
)
```

CHALLENGES

- formulas
- objects

'[airway][]' package.

- Himes BE, Jiang X, Wagner P, Hu R, Wang Q, Klanderman B, Whitaker
  RM, Duan Q, Lasky-Su J, Nikolos C, Jester W, Johnson M, Panettieri R
  Jr, Tantisira KG, Weiss ST, Lu Q. 'RNA-Seq Transcriptome Profiling
  Identifies CRISPLD2 as a Glucocorticoid Responsive Gene that
  Modulates Cytokine Function in Airway Smooth Muscle Cells.' PLoS
  One. 2014 Jun 13;9(6):e99625. PMID: 24926665. GEO: GSE52778.
  
[airway]: https://bioconductor.org/packages/airway
[GSE52778]: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52778

# We are very demanding of students

- Biological context
- Experimental protocol
- Statistical concepts
- Bioinformatic analysis

# Minimize the cost of initial analysis

```{r}
colData <- read.csv("~/extdata/rnaseq/colData.csv", row.names=1)
colData
```

```{r}
counts <- read.csv("~/extdata/rnaseq/counts.csv", row.names=1)
counts <- as.matrix(counts)
head(counts)
```

```{r}
design = ~ cell + dex
```

```{r, message=FALSE}
library(DESeq2)
dds <- DESeqDataSetFromMatrix(counts, colData, design)
```

```{r, message=FALSE}
fit <- DESeq(dds)
results <- results(fit)
```

```{r}
head(results)
```

# Present graphical representations with intuitive insight

'Biological' significance (fold change, x-axis) and 'statistical'
significance (P-value, y-axis)

```{r}
results[order(results$padj),]
plot(-log10(pvalue) ~ log2FoldChange, results)
```

# Revisit specific steps to develop understanding

## Visualization

Importance of cell line is readily apparent

```{r}
d <- dist(t(log(1 + counts)))
plot(cmdscale(d), col=colData$cell)
```

Distribution is strongly skewed(!)

```{r}
hist(counts[,1])
```

Mean / variance relationship is not consistent with simple Poisson model

```{r}
library(matrixStats)
plot(rowVars(counts) ~ rowMeans(counts), log = "xy")
abline(0, 1)
```

## Formula

```{r}
design = ~ cell + dex
```

Building understanding

```{r}
x <- rnorm(300)
df <- data.frame(x, y = 0.5 + x + rnorm(300))
plot(y ~ x, df)
fit1 <- lm(y ~ x, df)
abline(fit1, lwd=2, col="red")
```

QUESTION: regression through origin?

- add term, in this case a constant

```{r}
fit2 <- lm(y ~ 0 + x, df)
plot(y ~ x, df)
abline(fit1, col = "red")
abline(fit2, lwd = 2, col = "blue")
```

## Library size

Simple calculation

```{r}
colSums(counts)
```

Distribution, e.g., of one sample

```{r}
hist(log(1 + counts[,1]))
```

Approaches to 'normalization'

- QUESTION: Problems with arithmetic mean?
- Solutions: log counts
