_Bioconductor_ (https://bioconductor.org) is a flexible, widely used, and respected collection of _R_ packages for the statistical analysis and comprehension of many common types (bulk and single-cell sequencing, methylation and other microarrays, flow cytometry, ...) of high-throughput genomic data. Learning to use _Bioconductor_'s core infrastructure, domain-specific analysis packages, and annotation resources can pose significant challenges, both to those embarking on their first significant bioinformatic analysis and to those encountering the _Bioconductor_ ecosystem after developing considerable skill in other programming paradigms.

This workshop introduces strategies for effectively training users new to _Bioconductor_. Topics covered include the following

1. Assessing abilities: where do we begin? (9:00 - 9:45)

    Introduction. Course objectives. Course content.

    Biology - statistics - computation skills. Experience with _R_.
    
    _R_ refresher. Atomic vectors. Functions. Classes and
    methods. Help. Packages.

2. Engaging new and experienced users (9:45 - 10:30)

    Data input and manipulation in base _R_. Tidy data.
    
    _Bioconductor_. Bioinformatic data types and input. Genomic ranges.

BREAK (10:30 - 10:45)

3. Practical activities with meaningful feedback (10:45 - 11:30)

    _Bioconductor_ 'TxDb' and 'org' packages. Annotating ChIP-seq
    peaks. Scripts and functions.

4. Group activity (11:30 - 12:15)

LUNCH (12:15 - 1:15)

5. Managing cognitive load (1:15 - 2:00)

    RNA-seq differential expression. Count matrix. Experimental
    design. 'Top table'.
    
6. Matching expectations with learning objectives (2:15 - 3:00)

    Enabling independent learning. Package discovery and assessment.

BREAK (3:00 - 3:15)

7. Effective instruction and resources (3:15 - 4:00)

    Configured servers versus user laptops.

    Using _RStudio_. Markdown vignettes. Course packages. _Jupyter_
    notebooks.
    
    Best practices, literate programming, version control.
    
8. Group activity (4:00 - 4:45)

9. What have we learned? (4:45 - 5:00)

Acknowledgements
